pkg/parser/epub/metadata.go

```diff
@@ -588,6 +634,149 @@ func (m PubMetadataAdapter) LocalizedSortAs() *manifest.LocalizedString {
+		for _, item := range items {
+			if profile := a11yProfile(item.value); profile != "" {
+				profiles = append(profiles, profile)
+			}
+		}
+	}
+
+	for _, link := range m.Links(VocabularyDCTerms + "conformsTo") {
+		if profile := a11yProfile(link.href); profile != "" {
+			profiles = append(profiles, profile)
+		}
+	}
+
+	return profiles
+}
+
+func a11yProfile(value string) manifest.A11yProfile {
+	switch value {
+	case "EPUB Accessibility 1.1 - WCAG 2.0 Level A",
+		"http://idpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
+		"http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
+		"https://idpf.org/epub/a11y/accessibility-20170105.html#wcag-a",
+		"https://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-a":
+		return manifest.EPUBA11y10WCAG20A
+
+	case "EPUB Accessibility 1.1 - WCAG 2.0 Level AA",
+		"http://idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa",
+		"http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa",
+		"https://idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa",
+		"https://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa":
+		return manifest.EPUBA11y10WCAG20AA
+
+	case "EPUB Accessibility 1.1 - WCAG 2.0 Level AAA",
+		"http://idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa",
+		"http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa",
+		"https://idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa",
+		"https://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa":
+		return manifest.EPUBA11yWCAG20AAA
+
+	default:
+		return ""
+	}
+}
+
+func (m PubMetadataAdapter) a11yCertification() *manifest.A11yCertification {
+	certifierItem, _ := m.First(VocabularyA11Y + "certifiedBy")
+	c := manifest.A11yCertification{
+		CertifiedBy: certifierItem.value,
+	}
+
+	if certifierItem.id != "" {
+		if items, ok := certifierItem.children[VocabularyA11Y+"certifierCredential"]; ok && len(items) > 0 {
+			c.Credential = items[0].value
+		}
+		if link, ok := m.FirstLinkRefining(VocabularyA11Y+"certifierReport", certifierItem.id); ok {
+			c.Report = link.href
+		}
+	} else {
+		c.Credential = m.FirstValue(VocabularyA11Y + "certifierCredential")
+		c.Report = m.FirstValue(VocabularyA11Y + "certifierReport")
+		if c.Report == "" {
+			if link, ok := m.FirstLink(VocabularyA11Y + "certifierReport"); ok {
+				c.Report = link.href
+			}
+		}
+	}
+
+	if c.IsEmpty() {
+		return nil
+	}
+	return &c
+}
+
+func (m PubMetadataAdapter) a11ySummary() string {
+	return m.FirstValue(VocabularySchema + "accessibilitySummary")
+}
+
+func (m PubMetadataAdapter) a11yAccessModes() []manifest.A11yAccessMode {
+	am := []manifest.A11yAccessMode{}
+	for _, v := range m.Values(VocabularySchema + "accessMode") {
+		am = append(am, manifest.A11yAccessMode(v))
+	}
+	return am
+}
+
+func (m PubMetadataAdapter) a11yAccessModesSufficient() [][]manifest.A11yPrimaryAccessMode {
+	ams := [][]manifest.A11yPrimaryAccessMode{}
```

Comment:
Same optimization for allocation can be made

Reply:
This one is not easy because we ignore empty lists, so we don't know in advance the length of the array.

Reply:
Yes, but it doesn't really matter as long as we're close enough. It's better to over-allocate once rather than repeatedly allocate, and the amount we're over-allocating by is very small anyway. By doing `make(X, 0, len(Y))` we're just growing the capacity of the slice, but not the size, so if we end up overestimating the capacity, the length of the slice is still correct

Reply:
Alright, let me update it.